test/testdata/resolver/recursive_mixes_in_class_methods.rb

```diff
@@ -0,0 +1,121 @@
+# typed: true
+
+# Test no recursion for T::Helpers#mixes_in_class_methods
+
+module NoRecursion
+  module Foo
+    extend T::Helpers
+    module ClassMethods
+      def foo; end
+    end
+    mixes_in_class_methods(ClassMethods)
+    def instance_method; end
+    def self.class_method; end
+  end
+
+  module Bar
+    extend T::Helpers
+    include Foo
+    module ClassMethods
+      def bar1; end
+    end
+    module ClassMethods
+      def bar2; end
+    end
+    mixes_in_class_methods(ClassMethods)
+  end
+
+  module Baz
+    extend T::Helpers
+    include Bar
+    module ClassMethods
+      def baz; end
+    end
+    mixes_in_class_methods(ClassMethods)
+  end
+
+  class Qux
+    include Baz
+  end
+
+  Qux.foo            # error: Method `foo` does not exist on `T.class_of(NoRecursion::Qux)`
+  Qux.bar1           # error: Method `bar1` does not exist on `T.class_of(NoRecursion::Qux)`
+  Qux.bar2           # error: Method `bar2` does not exist on `T.class_of(NoRecursion::Qux)`
+  Qux.baz
+  Qux.new.instance_method
+  Qux.class_method   # error: Method `class_method` does not exist on `T.class_of(NoRecursion::Qux)`
```

Comment:
For my own understanding, a question:

The behavior here in this test is the same behavior that we would expect to see at runtime if the module hierarchy had been set up dynamically with

```ruby
def self.included(other)
  super
  other.extend(ClassMethods)
end
```

instead of `mixes_in_class_methods(ClassMethods)`, right?

(The original goal of `mixes_in_class_methods` was to model the `other.extend` pattern that occurs in Ruby code, so that you could replace one thing for the other with no change in behavior. This is [documented here](https://sorbet.org/docs/abstract#interfaces-and-the-included-hook).)